programs/zstdcli.c

```diff
@@ -1240,25 +1242,37 @@ int main(int const argCount, const char* argv[])
         CLEAN_RETURN(1);
     }
 
+    if (patchFromDictFileName != NULL && filenames->tableSize > 1) {
+        DISPLAY("error : can't use --patch-from=# on multiple files \n");
+        CLEAN_RETURN(1);
+    }
+
     /* No status message in pipe mode (stdin - stdout) or multi-files mode */
     if (!strcmp(filenames->fileNames[0], stdinmark) && outFileName && !strcmp(outFileName,stdoutmark) && (g_displayLevel==2)) g_displayLevel=1;
     if ((filenames->tableSize > 1) & (g_displayLevel==2)) g_displayLevel=1;
 
     /* IO Stream/File */
     FIO_setNotificationLevel(g_displayLevel);
     FIO_setPatchFromMode(prefs, patchFromDictFileName != NULL);
-    if (patchFromDictFileName != NULL) {
-        dictFileName = patchFromDictFileName;
-    }
     if (memLimit == 0) {
         if (compressionParams.windowLog == 0) {
             memLimit = (U32)1 << g_defaultMaxWindowLog;
         } else {
             memLimit = (U32)1 << (compressionParams.windowLog & 31);
     }   }
-    FIO_setMemLimit(prefs, memLimit);
+    if (patchFromDictFileName != NULL)
+        dictFileName = patchFromDictFileName;
     if (operation==zom_compress) {
 #ifndef ZSTD_NOCOMPRESS
+        if (patchFromDictFileName != NULL) {
+            const char* const srcFileName = filenames->fileNames[0];
```

Comment:
Could this logic be encapsulated into `fileio.c` ?

We try to reduce the amount of logic within `zstdcli.c` to a minimum, 
mostly limited to just parsing command line.